src/app-router.js

```diff
@@ -0,0 +1,43 @@
+import { Routes, Route } from "react-router-dom";
+import LoginPage from "./pages/login";
+import SignUpPage from "./pages/singup";
+import WelComePage from "./pages/welcome";
+import TestPage from "./pages/test";
+import ProductList from "./pages/product-list";
+import Cart from "./pages/cart";
+import Wishlist from "./pages/wishlist";
+import ProductInfo from "./pages/product-list/Components/ProductInfo";
+
+const NoMatch = () => <h3>404 - No matching route found</h3>;
+
+const isAuthenticated = true;
+
+const publicRoutes = [
+  { path: "/", name: <WelComePage /> },
+  { path: "/home", name: <ProductList /> },
+  { path: "/signup", name: <SignUpPage /> },
+  { path: "/login", name: <LoginPage /> },
+  { path: "/test", name: <TestPage /> },
+  { path: "*", name: <NoMatch /> },
+];
+
+const privateRoutes = [
+  { path: "/product-list", name: <ProductList /> },
```

Comment:
As you are storing routes in an array and then mapping over them, see  if `useroutes` https://reactrouter.com/docs/en/v6/api#useroutes can help you any how.

If you have not explored that, so you can.

Reply:
nice suggestion